lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_send_app_size_metrics.rb

```diff
@@ -0,0 +1,165 @@
+require 'plist'
+require_relative '../../helper/app_size_metrics_helper'
+
+module Fastlane
+  module Actions
+    class IosSendAppSizeMetricsAction < Action
+      def self.run(params)
+        # Check input parameters
+        api_url = URI(params[:api_url])
+        api_token = params[:api_token]
+        if (api_token.nil? || api_token.empty?) && !api_url.is_a?(URI::File)
+          UI.user_error!('An API token is required when using an `api_url` with a scheme other than `file://`')
+        end
```

Comment:
Have you considered making the API token an `optional: false` parameter?

From a consumer point of view, we don't expect users to call this action with a `file://` API URL, but always and only with an `https://` URL plus a token. We could ignore the token in the `file://` case here, which is what the `send_metrics` method seems to do, too.

Reply:
I'm not sure I'm following your argument here 😅 

> we don't expect users to call this action with a file:// API URL

Which is why it has to be `optional: true`, otherwise fastlane itself will `user_error!` all by itself if you provide a `file://` URL but no `api_token` (and why would you provide an API token when you're using `file://` URL, and what would that token be anyway… be it for debugging while we don't yet have a metrics server and thus any token to provide, or if you use this action to store the metrics as artefacts on your CI rather than sending them to a metrics server)

So, yes, the `token` in the `file://` case is ignored here (that's exactly what those lines 11–13 are for, to ignore it and allow it to be optional provided… **_only_** if using a file URL, and be required **_only_** when using a non-file URL). 
Given it has a different optionality depending on the type of `api_url` used, fastlane won't let us use `optional: false` in the `ConfigItem` otherwise it will complain that it's missing regardless of if it was valid to omit it (with `file://` urls) or not. So the only way in custom fastlane actions to provide conditional optionality is to handle it ourselves, as `ConfigItem` does not provide a way for that directly.

Reply:
> So imho while it definitively won't be the majority of case

Okay, cool. If that's the expectation, then it makes sense to keep the API consistent for both, rather than optimize it only for a scenario.

👍